language/internal/cc/macros.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2025 EngFlow Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cc
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// List of defined/known macro definition and their corresponding integer values, e.g {"__ANDROID__": 1, "_M_ARM": 1}
+// Any defined macro definition that does not have explicit value, is assumed to be equal 1, eg. `_WIN32`: 1
+// We don't support string/float macro definitions and using them in comparsion expressions
+type Macros map[string]int // e.g.
+
+type MacroDefinition struct {
```

Comment:
Naming nit: if you move this package to `language/internal/cc/macro`, you can drop "Macro" from a lot of the names: `macro.Definition`, `macro.Parse` and so on.